hswindow/hsdialog/hsprojectpropertiesdialog/projectpropertiesdialog.go

```diff
@@ -73,14 +72,10 @@ func (p *ProjectPropertiesDialog) Show(project *hsproject.Project, config *hscon
 	p.Dialog.Show()
 }
 
-func (p *ProjectPropertiesDialog) Render() {
-	if !p.Visible {
-		return
-	}
-
+func (p *ProjectPropertiesDialog) Build() {
 	canSave := len(strings.TrimSpace(p.project.ProjectName)) > 0
 
-	hswidget.ModalDialog("Select Auxiliary MPQ##ProjectPropertiesSelectAuxMPQDialog", &p.mpqSelectDialogVisible, g.Layout{
+	p.IsOpen(&p.mpqSelectDialogVisible).Layout(g.Layout{
 		g.Child("ProjectPropertiesSelectAuxMPQDialogLayout").Size(300, 200).Layout(g.Layout{
 			g.ListBox("ProjectPropertiesSelectAuxMPQDialogItems", p.auxMPQNames).Border(false).OnChange(func(selectedIndex int) {
 				p.mpqSelectDlgIndex = selectedIndex
```

Comment:
we probably should rename `giu.PopupModal.IsOpen` to something else. I _think_ that it's using the input argument as the control value for whether or not the modal widget is "open". Maybe "OpenWith" or "SetOpenControl" or something. @essial what do you think?

Reply:
The only argument I can think of in favor if keeping it like that is that it's some kind of weird immediate mode factory constructor setup thing, but it doesn't fit with standard golang names and I can see the confusion. I think something like `SetIsOpen` or `SetOpenControl` would be more clear, as well as changing the rest of those names, like `SetSize` and `SetLayout`